### ISSUE TITLE
oot2k - rebalanceBasket can be called before settleRewardsInt to maximize rewards

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-01-derby-judging#201]
oot2k

high

# rebalanceBasket can be called before settleRewardsInt to maximize rewards

## Summary
The user can rebalance his Basket before the settleRewardsInt is called. This will give the user the ability to change his allocation of tokens to receive max amount of rewards, in expense of others.
## Vulnerability Detail
https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/Game.sol#L506
This comment states that the rebalance should not be possible before settleRewardsInt is called.
But after step 6 is executed the rebalanceBasket function can be called.
An attacker could  frontrun rebalanceBasket setting his allocations to the highest rewarded protocols.
After settleRewards has been called, he can rebalance his Basket again to gain rewards.

## Impact
The user can always get the best reward without having to stake his tokens right in the first place.
This is in the expense of others.
## Code Snippet
https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/Game.sol#L506
https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/Game.sol#L324
https://github.com/sherlock-audit/2023-01-derby/blob/main/derby-yield-optimiser/contracts/Game.sol#L476
rebalanceBasket only checks for rebalancing of the vaults, after step 6 finishes this check should pass.
 
## Tool used

Manual Review

## Recommendation
Do not allow to call rebalanceBasket till Step 8 has finished. 

Duplicate of #259
